tests/components/nest/camera_sdm_test.py

```diff
@@ -0,0 +1,165 @@
+DEVICE_ID = "some-device-id"
+
+
+class FakeResponse:
+    """A fake web response used for returning results of commands."""
+
+    def __init__(self, json):
+        """Initialize the FakeResponse."""
+        self._json = json
+
+    def raise_for_status(self):
+        """Mimics a successful response status."""
+        pass
+
+    async def json(self):
+        """Return a dict with the response."""
+        return self._json
+
+
+class FakeAuth(AbstractAuth):
+    """Fake authentication object that returns fake responses."""
+
+    def __init__(self, response: FakeResponse):
+        """Initialize the FakeAuth."""
+        super().__init__(None, "")
+        self._response = response
+
+    async def async_get_access_token(self):
+        """Return a fake access token."""
+        return "some-token"
+
+    async def creds(self):
+        """Return a fake creds."""
+        return None
+
+    async def request(self, method: str, url: str, **kwargs):
+        """Pass through the FakeResponse."""
+        return self._response
+
+
+async def async_setup_camera(hass, traits={}, auth=None):
+    """Set up the platform and prerequisites."""
+    devices = {}
+    if traits:
+        devices[DEVICE_ID] = Device.MakeDevice(
+            {
+                "name": DEVICE_ID,
+                "type": CAMERA_DEVICE_TYPE,
+                "traits": traits,
+            },
+            auth=auth,
+        )
+    return await async_setup_sdm_platform(hass, PLATFORM, devices)
+
+
+async def test_no_devices(hass):
+    """Test configuration that returns no devices."""
+    await async_setup_camera(hass)
+    assert len(hass.states.async_all()) == 0
+
+
+async def test_ineligible_device(hass):
+    """Test configuration with devices that do not support cameras."""
+    await async_setup_camera(
+        hass,
+        {
+            "sdm.devices.traits.Info": {
+                "customName": "My Camera",
+            },
+        },
+    )
+    assert len(hass.states.async_all()) == 0
+
+
+async def test_camera_device(hass):
+    """Test a basic camera with a live stream."""
+    await async_setup_camera(
+        hass,
+        {
+            "sdm.devices.traits.Info": {
+                "customName": "My Camera",
+            },
+            "sdm.devices.traits.CameraLiveStream": {
+                "maxVideoResolution": {
+                    "width": 640,
+                    "height": 480,
+                },
+                "videoCodecs": ["H264"],
+                "audioCodecs": ["AAC"],
+            },
+        },
+    )
+
+    assert len(hass.states.async_all()) == 1
+    camera = hass.states.get("camera.my_camera")
+    assert camera is not None
+    assert camera.state == STATE_IDLE
+
+    registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = registry.async_get("camera.my_camera")
+    assert entry.unique_id == "some-device-id-camera"
+    assert entry.original_name == "My Camera"
+    assert entry.domain == "camera"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)